partiql-planner/src/testFixtures/resources/inputs/basics/case.sql

```diff
@@ -292,6 +292,100 @@ CASE t_item.t_string
     ELSE t_item.t_any
 END;
 
+--#[case-when-35]
+-- type: decimal(10,5)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE null
+END;
+
+--#[case-when-36]
+-- type: decimal(10,5)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE t_item.t_decimal_5_3
+END;
+
+--#[case-when-37]
+-- type: varchar(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE null
+END;
+
+--#[case-when-38]
+-- type: varchar(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE t_item.t_varchar_5
+END;
+
+--#[case-when-39]
+-- type: char(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE null
+END;
+
+--#[case-when-40]
+-- type: char(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE t_item.t_char_5
+END;
+
+--#[case-when-41]
+-- type: varchar(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE t_item.t_varchar_10
+END;
+
+--#[case-when-42]
+-- type: varchar(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE t_item.t_varchar_5
+END;
+
+--#[case-when-43]
+-- type: decimal
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE t_item.t_decimal
+END;
+
+--#[case-when-44]
+-- type: decimal
+-- coercion required. Common super type is arbitrary decimal
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE t_item.t_int32
```

Comment:
(observation) seems somewhat interesting that the common supertype here goes all the way to arbitrary decimals. It's correct but wondering if we could return a non-arbitrary precision type?

Perhaps we should only add that logic if there's a particular use case.

Reply:
Yeah I agree. 

Spent some time thinking about this actually. 

So INT-32 -> 2,147,483,647 -> which all can be hold under decimal(10,0).

So in case of `<int32> || <decimal(10,5)>`, we can return `decimal(10,5)` , 

and  in case of `<int32> || <decimal(5,3)>` we can return `decimal(10,3)`

to ensure no loss of precision. 

But upon second thought this appears to be "too smart" and a decision we might need spending more time on.

Reply:
Yeah, agree. Let's just keep as-is. If a use-case arises for the more precise logic, we can decide then.